### PR TITLE
Rename a lot of "awards" to "offers" [Finishes #156031299]

### PR DESCRIPTION
**Pull request by akostibas**
## Description

Amie helped us realize that the Award Queue was really more of an _offer_ queue, though it is still called the Award Queue by all the back office people.

The backend representations make a lot more sense if they're called "offers," so I've renamed quite a few functions and the database table `shipment_awards` to `shipment_offers`. I didn't rename everything because it is known by the name "Award Queue."

Hopefully this makes things make _more_ sense, not less.

## Reviewer Notes

* Keep an eye out for other functions or variables which might also benefit from the "award" -> "offer" renaming.

## Verification Steps

* [ ] All tests pass.
* [ ] The requirements listed in [Querying the Database Safely](https://github.com/transcom/mymove/blob/master/docs/backend.md#querying-the-database-safely) have been satisfied.

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/156031299) for this change

### CODE REVIEW

File: pkg/awardqueue/awardqueue.go [hunk 1]
```diff
@@ -25,16 +32,16 @@ type AwardQueue struct {
 	logger *zap.SugaredLogger
 }
 
-func (aq *AwardQueue) findAllUnawardedShipments() ([]models.PossiblyAwardedShipment, error) {
-	shipments, err := models.FetchUnawardedShipments(aq.db)
+func (aq *AwardQueue) findAllUnassignedShipments() ([]models.PossiblyAwardedShipment, error) {
+	shipments, err := models.FetchShipments(aq.db, true)
```
Comment (kilbergr) [inline pkg/awardqueue/awardqueue.go]: Using this boolean is a good way to combo the two fetch functions. However, I think it's opaque to call with just true passed in. I don't know whether it's good policy to pass in a var set to a bool that you've created exclusively for this purpose. Is there a better solution you know of?

File: pkg/awardqueue/awardqueue_test.go [hunk 2]
```diff
@@ -72,15 +72,15 @@ func (suite *AwardQueueSuite) Test_CheckShipmentDuringBlackOut() {
 	shipment, _ := testdatagen.MakeShipment(suite.db, time.Now(), time.Now(), time.Now().AddDate(0, 0, 1), tdl)
 
 	// Run the Award Queue
-	queue.assignUnawardedShipments()
+	queue.assignShipments()
 
-	shipmentAward := models.ShipmentAward{}
+	shipmentAward := models.ShipmentOffer{}
```
Comment (kilbergr) [inline pkg/awardqueue/awardqueue_test.go]: You don't want to change this var (and following similar vars) to shipmentOffer as well?

File: pkg/awardqueue/awardqueue_test.go [hunk 3]
```diff
@@ -175,7 +175,7 @@ func (suite *AwardQueueSuite) Test_AwardSingleShipment() {
 	}
 
 	// Run the Award Queue
-	award, err := queue.attemptShipmentAward(pas)
+	award, err := queue.attemptShipmentOffer(pas)
```
Comment (kilbergr) [inline pkg/awardqueue/awardqueue_test.go]: Change this award to offer?

File: pkg/models/shipment_offer_test.go [hunk 4]
```diff
@@ -8,7 +8,7 @@ import (
 )
 
 func (suite *ModelSuite) Test_ShipmentAwardValidations() {
-	sa := &ShipmentAward{}
+	sa := &ShipmentOffer{}
```
Comment (kilbergr) [inline pkg/models/shipment_offer_test.go]: here

File: pkg/testdatagen/make_awarded_shipment_record.go [hunk 5]
```diff
@@ -14,10 +14,10 @@ import (
 // MakeShipmentAward a single AwardedShipment record
 func MakeShipmentAward(db *pop.Connection, shipment models.Shipment,
 	tsp models.TransportationServiceProvider, admin bool, accepted *bool,
-	rejectionReason *string) (models.ShipmentAward, error) {
+	rejectionReason *string) (models.ShipmentOffer, error) {
 
 	// Add one awarded shipment record using existing shipment and TSP IDs
-	shipmentAward := models.ShipmentAward{
+	shipmentAward := models.ShipmentOffer{
```
Comment (kilbergr) [inline pkg/testdatagen/make_awarded_shipment_record.go]: here

File: pkg/awardqueue/awardqueue.go (hunk 1 above)
Comment (akostibas) [reply pkg/awardqueue/awardqueue.go]: I've been noticing this trend in general, specifically around our test data functions (`MakeTSP("good tsp", 1, true, tdl, Time.Now(), Time.Now())` as a not-too-made-up example, as we add more fields to each model).

Part of this is configuring our tools better to let us know what the parameters for functions are. Outside of that, I don't think that this one flag is worth creating some sort of fancier "options dictionary." If it grows, it probably would be.

**Comment by akostibas** [comment]
So much renaming has been done!

@kilbergr ready for another pass. I'm pretty sure I got everything.

File: pkg/awardqueue/awardqueue.go [hunk 6]
```diff
@@ -70,31 +77,31 @@ func (aq *AwardQueue) attemptShipmentAward(shipment models.PossiblyAwardedShipme
...
 
-				shipmentAward, err = models.CreateShipmentAward(aq.db, shipment.ID, tsp.ID, isAdministrativeShipment)
+				shipmentOffer, err = models.CreateShipmentOffer(aq.db, shipment.ID, tsp.ID, isAdministrativeShipment)
 				if err == nil {
-					if err = models.IncrementTSPPerformanceAwardCount(aq.db, tspPerformance.ID); err == nil {
+					if err = models.IncrementTSPPerformanceOfferCount(aq.db, tspPerformance.ID); err == nil {
 						if isAdministrativeShipment == true {
 							aq.logger.Info("Shipment pickup date is during a blackout period. Awarding Administrative Shipment to TSP.")
 						} else {
 							// TODO: AwardCount is off by 1
```
Comment (kilbergr) [inline pkg/awardqueue/awardqueue.go]: change to OfferCount is off by 1

**Pull request merged**